### PR TITLE
Add Fiber Debugger

### PR DESCRIPTION
**Pull request by gaearon**
# Fiber Debugger

This is a debugger handy for visualizing how [Fiber](https://github.com/facebook/react/issues/6170) works internally.

**It is only meant to be used by React contributors, and not by React users.**

It is likely that it might get broken at some point. If it's broken, ping [Dan](https://twitter.com/dan_abramov).

### Running

First, `npm run build` in React root repo folder.

Then `npm install` and `npm start` in this folder.

Open `http://localhost:3000` in Chrome.

### Features

* Edit code that uses `ReactNoop` renderer
* Visualize how relationships between fibers change over time
* Current tree is displayed in green

fiber debugger

### CODE REVIEW

File: src/renderers/noop/ReactNoop.js [hunk 1]
```diff
@@ -61,219 +62,231 @@ function flattenChildren(children : HostChildren<Instance | TextInstance>) {
   return flatArray;
 }
 
-var NoopRenderer = ReactFiberReconciler({
```
Comment (gaearon) [inline src/renderers/noop/ReactNoop.js]: These are just indentation changes + pass `options.debugTool` through to the reconciler.

File: src/renderers/shared/fiber/ReactFiberScheduler.js [hunk 2]
```diff
@@ -274,15 +275,27 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) {
...
     const current = workInProgress.alternate;
-    const next = beginWork(current, workInProgress, nextPriorityLevel);
 
-    if (next) {
-      // If this spawns new work, do that next.
-      return next;
-    } else {
-      // Otherwise, complete the current work.
-      return completeUnitOfWork(workInProgress);
+    if (debugTool) {
```
Comment (sebmarkbage) [inline src/renderers/shared/fiber/ReactFiberScheduler.js]: We should wrap all of these in `__DEV__` before we have the infrastructure in place to statically DCE them.

**Comment by gaearon** [comment]
@sebmarkbage Addressed.

**Comment by sebmarkbage** [comment]
ok!

**Pull request merged**